package lectures.week2oop


object Exceptions extends App {
  // создать исключение
  // throw new NullPointerException()

  //  try {
  //    intOrNothing(true)
  //  } catch {
  //    case e: RuntimeException => println("RTE is here")
  //  } finally {
  //    println("I will be there no matter what")
  //  }
  //
  //
  //}
  //
  //def intOrNothing(hasException: Boolean): Int = {
  //  if (hasException) throw new RuntimeException("Exception is here")
  //  else 200
  //
  println(Int.MaxValue)
  val x = 2147483647
  val y = 1


}



